core_lib_generator/TemplateCoreLib.yaml

```diff
@@ -0,0 +1,109 @@
+TestCoreLib:
+  env:
+    USER_DATA_USER: user
+    USER_DATA_PASSWORD: password
+    USER_DATA_PORT: 5432
+    USER_DATA_DB: user_data
+    USER_DATA_HOST: localhost
+    CUSTOMER_DATA_USER: user
+    CUSTOMER_DATA_PASSWORD: password
+    CUSTOMER_DATA_PORT: 5432
+    CUSTOMER_DATA_DB: customer_data
+    CUSTOMER_DATA_HOST: localhost
+    SOLR_HOST: localhost
+    SOLR_PORT: 8983
+    MEMCACHED_HOST: localhost
+    MEMCACHED_PORT: 11211
+  config:
+    data:
+      user_data:
+        log_queries: false
+        create_db: true
+        session:
+          pool_recycle: 3200
+          pool_pre_ping: false
+        url:
+          protocol: postgresql
+          username: ${oc.env:USER_DATA_USER}
+          password: ${oc.env:USER_DATA_PASSWORD}
+          host: ${oc.env:USER_DATA_HOST}
+          port: ${oc.env:USER_DATA_PORT}
+          file: ${oc.env:USER_DATA_DB}
+      customer_data:
+        log_queries: false
+        create_db: true
+        session:
+          pool_recycle: 3200
+          pool_pre_ping: false
+        url:
+          protocol: postgresql
+          username: ${oc.env:CUSTOMER_DATA_USER}
+          password: ${oc.env:CUSTOMER_DATA_PASSWORD}
+          host: ${oc.env:CUSTOMER_DATA_HOST}
+          port: ${oc.env:CUSTOMER_DATA_PORT}
+          file: ${oc.env:CUSTOMER_DATA_DB}
+      solr:
+        url:
+          protocol: https
+          host: ${oc.env:SOLR_HOST}
+          port: ${oc.env:SOLR_PORT}
+          file: solr/mycore
+    cache:
+      type: memcached
+      url:
+        host: ${oc.env:MEMCACHED_HOST}
+        port: ${oc.env:MEMCACHED_PORT}
+    jobs:
+      my_job:
+        initial_delay: 0s
+        frequency: ''
+        handler:
+          _target_: test_core_lib.core_lib.jobs.my_job.UpdateCustomerDetails
+  data_layers:
+    data:
+      user_data:
+        details:
+          name: details

```

Comment:
these comments refer to all data layers 
do we need a property called name, while the name shown under the grouping name?